scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -722,6 +722,17 @@ class TestSpsolveTriangular:
     def setup_method(self):
         use_solver(useUmfpack=False)
 
+    def test_zero_diagonal(self):
+        n = 5
+        A = np.random.randn(n, n)
```

Comment:
Use the new/recommended `Generator` interface
```suggestion
        rng = np.random.default_rng(43876432987)
        A = rng.standard_normal((n, n))
```